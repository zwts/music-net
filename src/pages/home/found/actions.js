import { getSearchResult } from '../../../service/neteaseCloudMusicApi';
import {
  FETCHING_FOUND_LIST,
  FETCH_FOUND_LIST_SUCCESS,
  FETCH_FOUND_LIST_FAILURE
} from '../../../redux/actionTypes';

export const fetchingFoundList = () => ({
  type: FETCHING_FOUND_LIST,
  loading: true
});

export const fetchFoundListSuccess = data => ({
  type: FETCH_FOUND_LIST_SUCCESS,
  loading: false,
  data: data
});

export const fetchFoundListFailure = error => ({
  type: FETCH_FOUND_LIST_FAILURE,
  loading: false
});

export const fetchFoundList = (art, limit) => {
  return dispatch => {
    dispatch(fetchingFoundList());
    return getSearchResult(art, limit).then(res => {
      const parsedList = parseFoundList(res);
      dispatch(fetchFoundListSuccess(parsedList));
    }).catch(error => dispatch(fetchFoundListFailure(error)));
  };
};

function parseFoundList(found) {
  const songsArray = found.data.result.songs;
  const parsedFoundList = [];
  songsArray.forEach(song => {
    const parsedSong = {};
    parsedSong.picUrl = song.al.picUrl;
    parsedSong.name = song.name;
    parsedSong.id = song.id;
    parsedSong.ar = song.ar[0].name;
    parsedFoundList.push(parsedSong);
  });
  dump(`parseFoundList: ${JSON.stringify(parsedFoundList)}`);
  return parsedFoundList;
}

