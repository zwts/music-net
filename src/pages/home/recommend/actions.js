import { getRecommendPlaylist } from "../../../service/neteaseCloudMusicApi";
import {
  REQUEST_RECOMMEND_LIST,
  RECEIVE_RECOMMEND_LIST
} from "../../../redux/actionTypes";

// Actions for recommend
export const requestRecommendList = () => ({
  type: REQUEST_RECOMMEND_LIST,
  loading: true
});

export const receiveRecommendList = data => ({
  type: RECEIVE_RECOMMEND_LIST,
  loading: false,
  data: data
});

export const fetchRecommendList = (limit) => {
  dump(`fetchRecommendList`);
  return dispatch => {
    dispatch(requestRecommendList());
    return getRecommendPlaylist(limit).then(result => {
      const parsedList = parseRecommendList(result);
      dispatch(receiveRecommendList(parsedList));
    });
  };
};

function parseRecommendList(recommendList) {
  dump(`parseRecommendList: ${JSON.stringify(recommendList)}`);
  const parsedList = [];
  recommendList.data.result.forEach(item => {
    const parsedItem = {};
    parsedItem.picUrl = item.picUrl;
    parsedItem.name = item.name;
    parsedItem.playCount = item.playCount;
    parsedItem.id = item.id;
    parsedList.push(parsedItem);
  });
  dump(`parsedList: ${JSON.stringify(parsedList)}`);
  return parsedList;
}
