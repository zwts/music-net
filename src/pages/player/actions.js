import {
  TOGGLE_PLAYER,
  FETCH_SONGURL_SUCCESS,
  CHANGE_PLAYER_MODE,
  FETCH_SONGURL_BEGIN,
  FETCH_SONGURL_FAILURE,
  PLAY_NEXT_SONG,
  PLAY_PREVIOUS_SONG,
  UPDATE_PLAYER_SONG_ID
} from "../../redux/actionTypes";
import { getSongUrl } from "../../service/neteaseCloudMusicApi";

export const togglePlayer = () => ({
  type: TOGGLE_PLAYER
});

export const fetchSongUrlBegin = () => ({
  type: FETCH_SONGURL_BEGIN
});

export const fetchSongUrlSuccess = songUrl => ({
  type: FETCH_SONGURL_SUCCESS,
  songUrl
});

export const fetchSongUrlFailure = error => ({
  type: FETCH_SONGURL_FAILURE,
  error
});

export const playNextSong = () => ({
  type: PLAY_NEXT_SONG,
});

export const playPreviousSong = () => ({
  type: PLAY_PREVIOUS_SONG,
});

export const changePlayerMode = playMode => ({
  type: CHANGE_PLAYER_MODE,
  mode: playMode
});

export const updatePlayerSongId = id => ({
  type: UPDATE_PLAYER_SONG_ID,
  songId: id
});

export function fetchSongUrl(id) {
  return dispatch => {
    dispatch(fetchSongUrlBegin());
    return  getSongUrl(id).then(res => {
      dispatch(fetchSongUrlSuccess(res.data.data[0].url));
      return res.data.data[0].url;
    })
      .catch(error => dispatch(fetchSongUrlFailure(error)));
  };
}


