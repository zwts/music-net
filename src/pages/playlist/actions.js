import { getPlaylistDetail } from "../../service/neteaseCloudMusicApi";
import {
  REQUEST_PLAYLIST,
  RECEIVE_PLAYLIST,
  UPDATE_PLAYLIST_INFO
} from "../../redux/actionTypes";

export const requestPlaylist = () => ({
  type: REQUEST_PLAYLIST,
  loading: true
});

export const receivePlaylist = data => ({
  type: RECEIVE_PLAYLIST,
  data: data,
  loading: false
});

export const updatePlaylistInfo = playlistInfo => ({
  type: UPDATE_PLAYLIST_INFO,
  info: playlistInfo
});

export const fetchPlaylist = (playlistId) => {
  dump(`fetchPlaylist ---`);
  return dispatch => {
    dispatch(requestPlaylist());
    return getPlaylistDetail(playlistId).then(result => {
      const parsedList = parsePlaylist(result);
      dispatch(receivePlaylist(parsedList));
    });
  };
};

function parsePlaylist(songs) {
  const songsArray = songs.data.playlist.tracks;
  dump(`parsePlaylist(): ${JSON.stringify(songsArray)}`);
  const parsedSongsList = [];
  songsArray.forEach(song => {
    const parsedSong = {};
    parsedSong.picUrl = song.al.picUrl;
    parsedSong.name = song.name;
    parsedSong.id = song.id;
    parsedSong.ar = song.ar[0].name;
    parsedSongsList.push(parsedSong);
  });
  dump(`parsePlaylist(): ${JSON.stringify(parsedSongsList)}`);
  return parsedSongsList;
}
