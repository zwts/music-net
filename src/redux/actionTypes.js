// Player panel actionType
export const TOGGLE_PLAYER = 'TOGGLE_PLAYER';
export const CHANGE_PLAYER_MODE = 'CHANGE_PLAYER_MODE';
export const FETCH_SONGURL_BEGIN = 'FETCH_SONGURL_BEGIN';
export const FETCH_SONGURL_SUCCESS = 'FETCH_SONGURL_SUCCESS';
export const FETCH_SONGURL_FAILURE = 'FETCH_SONGURL_FAILURE';
export const PLAY_NEXT_SONG = 'PLAY_NEXT_SONG';
export const PLAY_PREVIOUS_SONG = 'PLAY_NEXT_SONG';

// SongList
export const REQUEST_SONGLIST = 'REQUEST_SONGLIST';
export const RECEIVE_SONGLIST = 'RECEIVE_SONGLIST';

// Recommend
export const REQUEST_RECOMMEND_LIST = 'REQUEST_RECOMMEND_LIST';
export const RECEIVE_RECOMMEND_LIST = 'RECEIVE_RECOMMEND_LIST';

// Found
export const FETCHING_FOUND_LIST = 'FETCHING_FOUND_LIST';
export const FETCH_FOUND_LIST_SUCCESS = 'FETCH_FOUND_LIST_SUCCESS';
export const FETCH_FOUND_LIST_FAILURE = 'FETCH_FOUND_LIST_FAILURE';
