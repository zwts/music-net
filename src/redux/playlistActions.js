import { getRecommendPlaylist } from "../service/neteaseCloudMusicApi";
import {
  REQUEST_PLAYLIST,
  RECEIVE_PLAYLIST,
} from "./actionTypes";


export const requestPlaylist = () => ({
  type: REQUEST_PLAYLIST,
});

export const reqceivePlaylist = data => ({
  type: RECEIVE_PLAYLIST,
  data: data
});

export const fetchPlaylist = (limit) => {
  dump(`fetchPlaylist()`);
  return dispatch => {
    dispatch(requestPlaylist());
    return getRecommendPlaylist(limit).then(result => {
      const parsedList = parsePlaylist(result);
      dispatch(reqceivePlaylist(parsedList));
    });
  };
};


const samplePlaylist =
  {
    "data": {
      "hasTaste": false,
      "code": 200,
      "category": 0,
      "result": [{
        "id": 4926865057,
        "type": 0,
        "name": "夜间咖啡馆|西餐馆|傍晚的乐器演奏家",
        "copywriter": "编辑推荐：本单包含Bossa nova、民谣风、爵士乐、中世纪风格",
        "picUrl": "https://p1.music.126.net/Jh1iS5wFR5Xz_GNML996VA==/109951165046243126.jpg",
        "canDislike": false,
        "trackNumberUpdateTime": 1607590232764,
        "playCount": 83431,
        "trackCount": 28,
        "highQuality": false,
        "alg": "featured"
      }, {
        "id": 5097412008,
        "type": 0,
        "name": "网郁云 “那晚我悲痛欲绝”",
        "copywriter": "热门歌单推荐",
        "picUrl": "https://p1.music.126.net/2YOC31gtmRjoB-K5FSOE_w==/109951165520586580.jpg",
        "canDislike": true,
        "trackNumberUpdateTime": 1607996811286,
        "playCount": 946498,
        "trackCount": 260,
        "highQuality": false,
        "alg": "hot_server"
      }]
    }
  };


function parsePlaylist(playlist) {
  dump(`parsePlaylist(): ${JSON.stringify(playlist)}`);
  const parsedPlaylist = [];
  playlist.data.result.forEach(item => {
    const parsedItem = {};
    parsedItem.picUrl = item.picUrl;
    parsedItem.name = item.name;
    parsedItem.playCount = item.playCount;
    parsedItem.id = item.id;
    parsedPlaylist.push(parsedItem);
  });
  dump(`parsedPlaylist: ${JSON.stringify(parsedPlaylist)}`);
  return parsedPlaylist;
}

function parseSongsList(songsList) {
  const songsArray = songsList.data.result.songs;
  const parsedSongsList = [];
  songsArray.forEach(song => {
    const parsedSong = {};
    parsedSong.name = song.name;
    parsedSong.id = song.id;
    parsedSong.ar = song.ar[0].name;
    parsedSongsList.push(parsedSong);
  });
  dump(`parseSongsList(): ${JSON.stringify(parsedSongsList)}`);
  return parsedSongsList;
}
