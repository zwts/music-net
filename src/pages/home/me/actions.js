import { UPDATE_FAVORITE_SONG } from "../../../redux/actionTypes";
// const FAVORITE_SONG_LIMIT = 10;

export const updateFavoriteSong = (favSongs) => ({
  type: UPDATE_FAVORITE_SONG,
  favoriteSongs: favSongs
});

export const toggleFavoriteSong = (song, favSongs) => {
  return (dispatch) => {
    dump(`toggleFavoriteSong id ${song.id}`);
    const id = song.id.toString();

    if (favSongs.has(id)) {
      dump(`toggleFavoriteSong delete ${id}`);
      favSongs.delete(id);
    } else {
      //Todo should limited fav
      // if (favSongs.size < FAVORITE_SONG_LIMIT) {
      dump(`toggleFavoriteSong set ${id}`);
      favSongs.set(id, song);
      // }
    }

    dispatch(updateFavoriteSong(favSongs));
  }
};
