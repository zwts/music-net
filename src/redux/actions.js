import { TOGGLE_PLAYER, CHANGE_PLAYER_MODE } from "./actionTypes";

export const togglePlayer = () => ({
  type: TOGGLE_PLAYER
});

export const changePlayerMode = playMode => ({
  type: CHANGE_PLAYER_MODE,
  mode: playMode
});
