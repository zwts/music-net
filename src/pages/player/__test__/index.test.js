import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { togglePlayer } from '../../../redux/actions'

import Player from '../index';

const mockStore = configureStore([]);

describe('<Player />', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      playedState: 'stop',
      loopMode: 'list'
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Player />
      </Provider>
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on enter key click', () => {
    let event = new CustomEvent ('keydown', {
      bubbles: true, cancelable: false
    });
    event.key = 'Enter';

    document.dispatchEvent(event);

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(togglePlayer);
  });
});
