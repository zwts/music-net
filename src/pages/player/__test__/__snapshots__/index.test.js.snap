// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders correctly 1`] = `
<div
  className="player"
  onKeyDown={[Function]}
  tabIndex="-1"
>
  <span
    className="p-pri"
  >
    list
  </span>
  <span
    className="p-sec"
  >
    play
  </span>
  <audio
    controls={true}
    id="player-audio"
  />
</div>
`;
